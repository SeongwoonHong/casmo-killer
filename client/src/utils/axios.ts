import axios from 'axios';

axios.interceptors.request.use((config) => {
  console.log(config);
  return config;
});

function setTokenToHeader(header, csrfToken) {
  axios.defaults.headers.common[header] = csrfToken;
}

export {
  axios,
  setTokenToHeader,
};
