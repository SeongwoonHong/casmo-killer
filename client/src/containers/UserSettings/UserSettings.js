import React, { Component } from 'react';
import axios from 'axios/index';

import * as storage from '@sharedUtils/storage';
import LoadingOverlay from '@sharedComponents/LoadingOverlay';
import FormMessage from '@sharedComponents/FormMessage';

import ProfileSettings from '../../components/ProfileSettings';
import SecuritySettings from '../../components/SecuritySettings';

import './UserSettings.scss';

class MyAccount extends Component {

  constructor(props) {

    super(props);

    this.state = {
      isLoading: false,
      isSuccess: false,
      message: ''
    };

  }

  componentDidMount() {

    const { token } = this.props.match.params;

    if (token) {
      this.verifyToken(token);
      this.props.history.replace('/user/settings');
    }

  }

  onProfileChange = async (userInfo) => {
    const user = await storage.set('ckUser', userInfo);
    this.props.loginSuccess(user);
  };

  verifyToken = async (token) => {

    this.setState({ isLoading: true });

    if (this.props.user.isLoggedIn) {

      try {

        const { data } = await axios.put('/api/user/update/email', { token });

        if (data && data.user && data.message) {

          this.setState({
            isSuccess: true,
            message: data.message
          });

          const user = await storage.set('ckUser', data.user);
          this.props.loginSuccess(user);

        } else {

          this.setState({
            isSuccess: false,
            message: 'Failed to communicate with the server.'
          });

        }

      } catch (error) {

        console.error(error);
        this.setState({
          isSuccess: false,
          message: error.response.data.message
        });

      }

    }

    this.setState({ isLoading: false });

  };

  render() {

    const { user } = this.props;
    const { isLoading, isSuccess, message } = this.state;

    return (
      <div className="My-account">
        <LoadingOverlay
          isVisible={ isLoading }
          overlayColor="rgba(0,0,0,.75)"
          circleColor="#fff" />
        <h2 className="user-page-title">
          My Account
          <i className="material-icons">
            person
          </i>
        </h2>
        <FormMessage
          message={ message }
          type={ isSuccess ? 'success' : 'error' } />
        <ProfileSettings
          user={ user }
          onSuccess={ this.onProfileChange } />
        <SecuritySettings
          strategy={ user.strategy } />
      </div>
    );

  }

}

export default MyAccount;
