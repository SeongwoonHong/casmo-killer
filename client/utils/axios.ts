import axios from 'axios';

const baseURL = 'http://localhost:9000';

axios.defaults.baseURL = baseURL + '/api';

export { axios };
