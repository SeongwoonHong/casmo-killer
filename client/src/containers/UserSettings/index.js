import { connect } from 'react-redux';
import * as actions from '@actions';

import UserSettings from './UserSettings';

const mapStateToProps = (state) => {
  return {
    auth: state.auth,
    user: state.user.user
  };
};

const mapDispatchToProps = (dispatch) => {
  return {
    loginSuccess: payload => dispatch(actions.loginSuccess(payload))
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(UserSettings);
