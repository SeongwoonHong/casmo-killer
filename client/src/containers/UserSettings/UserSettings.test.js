/* eslint react/jsx-boolean-value: 0 */
import React from 'react';
import { shallow } from 'enzyme';
import sinon from 'sinon';
import axios from 'axios';

import UserSettings from './UserSettings';

describe('<UserSettings />', () => {

  const userData = {
    strategy: 'local'
  };

  let component;
  let mockGet;
  let mockPut;

  beforeEach(() => {

    component = shallow(
      <UserSettings
        user={ userData }
        match={ { params: {} } } />
    );

    mockGet = sinon.stub(axios, 'get');
    mockPut = sinon.stub(axios, 'put');

  });

  afterEach(() => {
    mockGet.restore();
    mockPut.restore();
  });

  it('should render the component consistently', () => {
    expect(component).toMatchSnapshot();
  });

  it('should verify the token', () => {

    const paramData = { token: 'abcdeft' };

    const spy = sinon.spy(UserSettings.prototype, 'verifyToken');
    const replace = sinon.spy();

    shallow(
      <UserSettings
        user={ userData }
        history={ { replace } }
        match={ { params: paramData } } />
    );

    expect(spy.calledWith(paramData.token)).toEqual(true);
    expect(replace.calledWith('/user/settings')).toEqual(true);

  });

  it('should verify the token by making a request to the server', async () => {

    const setStateSpy = sinon.spy(UserSettings.prototype, 'setState');
    const spy = sinon.spy();

    const responseData = {
      data: {
        user: {
          isLoggedIn: true
        },
        message: 'server message'
      }
    };
    const promise = Promise.resolve(responseData);

    mockPut.callsFake(() => promise);

    shallow(
      <UserSettings
        user={ { isLoggedIn: true } }
        history={ { replace: () => {} } }
        loginSuccess={ spy }
        match={ { params: { token: 'abcdeft' } } } />
    );

    await promise;

    expect(setStateSpy.calledWith({
      isSuccess: true,
      message: responseData.data.message
    })).toEqual(true);

    // simulating async storage method
    await Promise.resolve({ isLoggedIn: true });

    expect(spy.calledWith({ isLoggedIn: true })).toEqual(true);

  });

});
