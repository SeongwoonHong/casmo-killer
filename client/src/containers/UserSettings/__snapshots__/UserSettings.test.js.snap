// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserSettings /> should render the component consistently 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MyAccount
    match={
        Object {
            "params": Object {},
          }
    }
    user={
        Object {
            "strategy": "local",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <LoadingOverlay
          circleColor="#fff"
          circleSize=""
          isVisible={false}
          overlayColor="rgba(0,0,0,.75)"
/>,
        <h2
          className="user-page-title"
>
          My Account
          <i
                    className="material-icons"
          >
                    person
          </i>
</h2>,
        <FormMessage
          message=""
          type="error"
/>,
        <ProfileSettings
          onSuccess={[Function]}
          user={
                    Object {
                              "strategy": "local",
                            }
          }
/>,
        <SecuritySettings
          strategy="local"
/>,
      ],
      "className": "My-account",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "circleColor": "#fff",
          "circleSize": "",
          "isVisible": false,
          "overlayColor": "rgba(0,0,0,.75)",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "My Account",
            <i
              className="material-icons"
>
              person
</i>,
          ],
          "className": "user-page-title",
        },
        "ref": null,
        "rendered": Array [
          "My Account",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "person",
              "className": "material-icons",
            },
            "ref": null,
            "rendered": "person",
            "type": "i",
          },
        ],
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "message": "",
          "type": "error",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "onSuccess": [Function],
          "user": Object {
            "strategy": "local",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "strategy": "local",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <LoadingOverlay
            circleColor="#fff"
            circleSize=""
            isVisible={false}
            overlayColor="rgba(0,0,0,.75)"
/>,
          <h2
            className="user-page-title"
>
            My Account
            <i
                        className="material-icons"
            >
                        person
            </i>
</h2>,
          <FormMessage
            message=""
            type="error"
/>,
          <ProfileSettings
            onSuccess={[Function]}
            user={
                        Object {
                                    "strategy": "local",
                                  }
            }
/>,
          <SecuritySettings
            strategy="local"
/>,
        ],
        "className": "My-account",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "circleColor": "#fff",
            "circleSize": "",
            "isVisible": false,
            "overlayColor": "rgba(0,0,0,.75)",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "My Account",
              <i
                className="material-icons"
>
                person
</i>,
            ],
            "className": "user-page-title",
          },
          "ref": null,
          "rendered": Array [
            "My Account",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "person",
                "className": "material-icons",
              },
              "ref": null,
              "rendered": "person",
              "type": "i",
            },
          ],
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "message": "",
            "type": "error",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "onSuccess": [Function],
            "user": Object {
              "strategy": "local",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "strategy": "local",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
